haystack/pipelines/base.py

```diff
@@ -987,18 +989,32 @@ def execute_eval_run(
 
             tracker.track_params({"pipeline_index_document_count": document_count})
 
-            eval_result = query_pipeline.eval(
-                labels=evaluation_set_labels,
-                params=query_params,
-                sas_model_name_or_path=sas_model_name_or_path,
-                sas_batch_size=sas_batch_size,
-                sas_use_gpu=sas_use_gpu,
-                add_isolated_node_eval=add_isolated_node_eval,
-                custom_document_id_field=custom_document_id_field,
-                context_matching_boost_split_overlaps=context_matching_boost_split_overlaps,
-                context_matching_min_length=context_matching_min_length,
-                context_matching_threshold=context_matching_threshold,
-            )
+            if use_batch_mode:
+                eval_result = query_pipeline.eval_batch(
+                    labels=evaluation_set_labels,
+                    params=query_params,
+                    sas_model_name_or_path=sas_model_name_or_path,
+                    sas_batch_size=sas_batch_size,
+                    sas_use_gpu=sas_use_gpu,
+                    add_isolated_node_eval=add_isolated_node_eval,
+                    custom_document_id_field=custom_document_id_field,
+                    context_matching_boost_split_overlaps=context_matching_boost_split_overlaps,
+                    context_matching_min_length=context_matching_min_length,
+                    context_matching_threshold=context_matching_threshold,
+                )
+            else:
+                eval_result = query_pipeline.eval(
+                    labels=evaluation_set_labels,
+                    params=query_params,
+                    sas_model_name_or_path=sas_model_name_or_path,
+                    sas_batch_size=sas_batch_size,
+                    sas_use_gpu=sas_use_gpu,
+                    add_isolated_node_eval=add_isolated_node_eval,
+                    custom_document_id_field=custom_document_id_field,
+                    context_matching_boost_split_overlaps=context_matching_boost_split_overlaps,
+                    context_matching_min_length=context_matching_min_length,
+                    context_matching_threshold=context_matching_threshold,
+                )
```

Comment:
These are the same parameters for both function calls - not sure if it makes sense to create a dict and then unpack the dict in the two method calls. Maybe that's a bad practice, but it makes the code more compact.

Reply:
Interesting thought. In this case here, I would leave it as is because it's part of the user-facing function pipeline.eval_batch that users can call directly and it occurs only twice here. I think listing all the parameters is more intuitive and easier to understand for users than having some dictionary/custom datastructure that they first need to understand. If it was a function that is used internally only or if it occurred much more often, we could make it more compact with your suggestion yes, I agree.

Reply:
I agree, good points.